packages/read-model/src/Product/Detail/DetailProductViewFacade.php

```diff
@@ -0,0 +1,120 @@
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductCachedAttributesFacade $productCachedAttributesFacade
+     * @param \Shopsys\FrameworkBundle\Model\Category\CategoryFacade $categoryFacade
+     * @param \Shopsys\ReadModelBundle\Image\ImageViewFacade $imageViewFacade
+     * @param \Shopsys\ReadModelBundle\Product\Action\ProductActionViewFacade $productActionViewFacade
+     * @param \Shopsys\ReadModelBundle\Brand\BrandViewFacade $brandViewFacade
+     * @param \Shopsys\ReadModelBundle\Parameter\ParameterViewFacade $parameterViewFacade
+     */
+    public function __construct(
+        DetailProductViewFactory $detailProductViewFactory,
+        ProductOnCurrentDomainFacadeInterface $productOnCurrentDomainFacade,
+        ProductCachedAttributesFacade $productCachedAttributesFacade,
+        CategoryFacade $categoryFacade,
+        ImageViewFacade $imageViewFacade,
+        ProductActionViewFacade $productActionViewFacade,
+        BrandViewFacade $brandViewFacade,
+        ParameterViewFacade $parameterViewFacade
+    ) {
+        $this->detailProductViewFactory = $detailProductViewFactory;
+        $this->productOnCurrentDomainFacade = $productOnCurrentDomainFacade;
+        $this->productCachedAttributesFacade = $productCachedAttributesFacade;
+        $this->categoryFacade = $categoryFacade;
+        $this->imageViewFacade = $imageViewFacade;
+        $this->productActionViewFacade = $productActionViewFacade;
+        $this->brandViewFacade = $brandViewFacade;
+        $this->parameterViewFacade = $parameterViewFacade;
+    }
+
+    /**
+     * @param int $productId
+     * @return \Shopsys\ReadModelBundle\Product\Detail\DetailProductView
+     */
+    public function getVisibleProductDetail(int $productId): DetailProductView
+    {
+        $product = $this->productOnCurrentDomainFacade->getVisibleProductById($productId);
+
+        return $this->createFromProduct($product);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Product $product
+     * @return \Shopsys\ReadModelBundle\Product\Detail\DetailProductView
+     */
+    protected function createFromProduct(Product $product): DetailProductView
+    {
+        $imageViews = $this->imageViewFacade->getAllImagesByEntityId(Product::class, $product->getId());
+        $productActionView = $this->productActionViewFacade->getForProduct($product);
+        $brandView = $this->brandViewFacade->getForProduct($product);
+        $parameterViews = $this->parameterViewFacade->getAllForProduct($product);
+
+        return $this->detailProductViewFactory->createFromProduct(
+            $product,
+            $this->productCachedAttributesFacade->getProductSellingPrice($product),
+            $this->categoryFacade->getProductMainCategoryOnCurrentDomain($product)->getId(),
+            $imageViews,
+            $brandView,
+            $productActionView,
+            $parameterViews
+        );
+    }
```

Comment:
To not have the factory dependent on the facade. Facades should be at the top of our architecture as we discussed earlier.
So basically to avoid having the necessity to have autowired facades in the factory for better data aggregation (= when I want prices from elasticsearch I rewrite facade instead of adding new methods into the factory)

Reply:
Discussed personally, I change implementation to avoid two createFromProduct methods and other unintuitive behavior.

Reply:
implemented in dec746716, b5f7a91ea and  a1e073bb7